From a57d363b18eaf327efa7a42fbe1c8de1339fb01b Mon Sep 17 00:00:00 2001
From: ckie <git-525ff67@ckie.dev>
Date: Wed, 13 Mar 2024 23:25:11 +0200
Subject: [PATCH] server/wivrn_session: Add fps field to header

---
 server/driver/wivrn_session.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/server/driver/wivrn_session.h b/server/driver/wivrn_session.h
index d1f30e4..b501dbb 100644
--- a/server/driver/wivrn_session.h
+++ b/server/driver/wivrn_session.h
@@ -63,6 +63,8 @@ class wivrn_session : public std::enable_shared_from_this<wivrn_session>
 
 	std::shared_ptr<audio_device> audio_handle;
 
+	float fps;
+
 	wivrn_session(TCP && tcp);
 
 public:
-- 
2.43.1

